calico-vpp-agent/cmd/calico_vpp_dataplane.go

```diff
@@ -98,28 +100,33 @@ func main() {
 		policyServer,
 		prometheusServer,
 		log.WithFields(logrus.Fields{"component": "cni"}),
+		&mainTomb,
 	)
 	if err != nil {
 		log.Errorf("Failed to create CNI server")
 		log.Fatal(err)
 	}
 
-	go routingServer.Serve()
+	mainTomb.Go(func() error { return routingServer.Serving() })
 	<-routing.ServerRunning
 
-	go policyServer.Serve()
+	mainTomb.Go(func() error { return policyServer.Serving() })
```

Comment:
Not a fan of the `Serving` terminology, it would be better to just call `Serve()` here and just create the error in this inline function.
e.g.
```
mainTomb.Go(func() error {
    policyServer.Serve() 
    return fmt.Errorf("Policy server exited")
})